File: src/main/java/eu/wejsonekk/bombocraft/commands/implementation/DiscordCommand.java
package eu.wejsonekk.bombocraft.commands.implementation;

import dev.rollczi.litecommands.annotations.command.Command;
import dev.rollczi.litecommands.annotations.context.Context;
import dev.rollczi.litecommands.annotations.execute.Execute;
import dev.rollczi.litecommands.annotations.permission.Permission;
import eu.wejsonekk.bombocraft.configuration.implementation.MessageConfiguration;
import eu.wejsonekk.bombocraft.notification.NotificationAnnouncer;
import lombok.Getter;
import org.bukkit.entity.Player;

@Command(name = "discord", aliases = {"dcserver", "teamspeak", "ts3", "dcserwer", "discordserwer"})
public class DiscordCommand {

    private final NotificationAnnouncer announcer;
    private final MessageConfiguration messages;

    public DiscordCommand(NotificationAnnouncer announcer, MessageConfiguration messages) {
        this.announcer = announcer;
        this.messages = messages;
    }

    @Execute
    void execute(@Context Player player){
        this.messages.standardCommands.DiscordCommand
                .stream().forEach(messages -> this.announcer.sendMessage(player, messages));
    }
    @Execute(name = "sync")
    @Permission("starblock.command.discord.sync")
    void executeSyncDiscord(@Context Player player){
        this.announcer.sendMessage(player, "not found message for common.discord.sync");
    }


}

File: src/main/java/eu/wejsonekk/bombocraft/commands/implementation/DiscordCommand.kt
package eu.wejsonekk.bombocraft.commands.implementation

import dev.rollczi.litecommands.annotations.command.Command
import dev.rollczi.litecommands.annotations.context.Context
import dev.rollczi.litecommands.annotations.execute.Execute
import dev.rollczi.litecommands.annotations.permission.Permission
import eu.wejsonekk.bombocraft.configuration.implementation.MessageConfiguration
import eu.wejsonekk.bombocraft.notification.NotificationAnnouncer
import org.bukkit.entity.Player

@Command(name = "discord", aliases = ["dcserver", "teamspeak", "ts3", "dcserwer", "discordserwer"])
class DiscordCommand(private val announcer: NotificationAnnouncer, private val messages: MessageConfiguration) {
    @Execute
    fun execute(@Context player: Player?) {
        messages.DiscordCommand
            .stream().forEach { messages: String? -> announcer.sendMessage(player, messages) }
    }

    @Execute(name = "sync")
    @Permission("starblock.command.discord.sync")
    fun executeSyncDiscord(@Context player: Player?) {
        announcer.sendMessage(player, "not found message for common.discord.sync")
    }
}
